internal/groups/groups.go

```diff
@@ -0,0 +1,111 @@
+package groups
+
+import (
+	"context"
+)
+
+type Member interface{}
+
+type Metadata map[string]interface{}
+
+type Group struct {
+	ID          string
+	OwnerID     string
+	ParentID    string
+	Name        string
+	Description string
+	Metadata    Metadata
+	Level       int
+	Path        string
+}
+
+type PageMetadata struct {
+	Total  uint64
+	Offset uint64
+	Limit  uint64
+	Name   string
+}
+
+type GroupPage struct {
+	PageMetadata
+	Groups []Group
+}
+
+type MemberPage struct {
+	PageMetadata
+	Members []Member
```

Comment:
Do we need Member type definition or `interface{}` is sufficient? I have no preferences, but we probably don't gain anything with this since we need to typecast anyways.